### ISSUE TITLE
Thrown a java.lang.NullPointerException for linked cluster

### ISSUE THREAD

**Issue by wezhang** [microsoft/azure-tools-for-java#1336]
```
null
java.lang.NullPointerException
	at com.microsoft.azure.hdinsight.serverexplore.hdinsightnode.ClusterNode.toProperties(ClusterNode.java:187)
	at com.microsoft.tooling.msservices.serviceexplorer.NodeActionListener.sendTelemetry(NodeActionListener.java:57)
	at com.microsoft.tooling.msservices.serviceexplorer.NodeActionListener.beforeActionPerformed(NodeActionListener.java:48)
	at com.microsoft.tooling.msservices.serviceexplorer.NodeAction.fireNodeActionEvent(NodeAction.java:59)
	at com.microsoft.intellij.components.ServerExplorerToolWindowFactory.treeMousePressed(ServerExplorerToolWindowFactory.java:138)
	at com.microsoft.intellij.components.ServerExplorerToolWindowFactory.access$100(ServerExplorerToolWindowFactory.java:73)
	at com.microsoft.intellij.components.ServerExplorerToolWindowFactory$1.mousePressed(ServerExplorerToolWindowFactory.java:99)
	at java.awt.AWTEventMulticaster.mousePressed(AWTEventMulticaster.java:280)
	at java.awt.Component.processMouseEvent(Component.java:6530)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at com.intellij.ui.treeStructure.Tree.processMouseEvent(Tree.java:423)
	at java.awt.Component.processEvent(Component.java:6298)
	at java.awt.Container.processEvent(Container.java:2236)
	at java.awt.Component.dispatchEventImpl(Component.java:4889)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4888)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4522)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4466)
	at java.awt.Container.dispatchEventImpl(Container.java:2280)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:90)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:822)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:646)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:366)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by wezhang** [comment]
For `HDInsightAdditionalClusterDetail`, `getSubscription()` method returns `null`

**Comment by wezhang** [comment]
fixed by #1333
